### PR TITLE
#312 adds basic corelib types to forge_print 

### PR DESCRIPTION
**Pull request by tkumor3**
[Make sure that forge_print works with all corelib basic types](https://github.com/foundry-rs/starknet-foundry/issues/312)
Closes #

## Introduced changes

-
-

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by piotmag769 — CHANGES_REQUESTED** [review]
I think we should implement print for signed types as well: i8, i16 etc.

**Comment by tkumor3** [comment]
There is no info in Cairo book about signed types 🤔. I found it in source code, so I added support for it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - `#[should_panic]` attribute support
 - Documentation to public methods
+- Supported data types `#[forge_print::PrintTrait]`
```
Comment (piotmag769) [inline CHANGELOG.md:16]:
```suggestion
- Added print support for basic numeric data types
```

**Review by piotmag769 — APPROVED** [review]

File: crates/forge/tests/e2e/running.rs [hunk 2]
```diff
@@ -240,8 +240,21 @@ fn with_print() {
...
+        original value: [123345678910]
+        original value: [0], converted to a string: []
+        original value: [10633823966279327296825105735305134080]
+        original value: [2], converted to a string: []
+        original value: [11], converted to a string: [
]
+        original value: [1234]
+        original value: [123456]
+        original value: [123456789]
+        original value: [12345612342]
```
Comment (war-in) [inline crates/forge/tests/e2e/running.rs:257]: Why do some of them have and some don't have the `converted to a string` part?

Comment (piotmag769) [reply crates/forge/tests/e2e/running.rs:257]: Depends if they are convertable to ascii - for example 1234 is 0x4D2, first byte is D2 = 210 > 127 (out of range of ascii)

**Review by war-in — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]
LGTM

**Pull request merged**